### PR TITLE
Avoid std::atomic_init deprecated by P0883.

### LINKED ISSUE #139: atomic_init calls throw deprecation warnings after acceptance of C++20 P0883
P0883 marks atomic_init as deprecated, which causes cppcoro to throw a bunch of deprecation errors when used with our standard library (post https://github.com/microsoft/STL/pull/390 ).

Should the places calling `atomic_init` be changed to do something like:

```
// earlier
#ifdef __cpp_lib_atomic_value_initialization
template<class T, class Args>
void true_placement_new(T* ptr, Args&&... args) { // do you already have a function for this?
    ::new(static_cast<void*>(ptr)) T(std::forward<Args>(args)...);
}
#endif // __cpp_lib_atomic_value_initialization


// in multi_producer_sequencer.hpp:
SEQUENCE seq = initialSequence - (bufferSize - 1);
do
{
#ifdef __cpp_lib_atomic_value_initialization
    true_placement_new(&m_published[seq & m_sequenceMask], seq);
#else
    std::atomic_init(&m_published[seq & m_sequenceMask], seq);
#endif // __cpp_lib_atomic_value_initialization
} while (seq++ != initialSequence);
```

instead?

### PR DESCRIPTION
**Pull request by BillyONeal**
Resolves #139

I don't think I've been successful in actually getting this to build; I apologize if there are typos here:

image

### CODE REVIEW

**Comment by lewissbaker** [comment]
> I don't think I've been successful in actually getting this to build;

The build system is currently only looking for VS2017 compilers.
You will need to pull in the vs2019 branch to get the build system to find the VS2019 compiler (See PR #125).

**Comment by BillyONeal** [comment]
>The build system is currently only looking for VS2017 compilers.

Right, I put the VS2017 compiler on the PATH. (That's the `-vcvars_ver=14.16.x` in the dev prompt command) But I guess it might be looking in the 2017 installation directory or something like that.

> This loop still needs to run to initialise the elements of the array.

Derp, stand by.

File: include/cppcoro/multi_producer_sequencer.hpp [hunk 1]
```diff
@@ -508,11 +508,13 @@ namespace cppcoro
 		constexpr unsigned_diff_t maxSize = static_cast<unsigned_diff_t>(std::numeric_limits<diff_t>::max());
 		assert(bufferSize <= maxSize);
 
+#ifndef __cpp_lib_atomic_value_initialization
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply include/cppcoro/multi_producer_sequencer.hpp]: Fixed up, thank you!

**Comment by BillyONeal** [comment]
FYI, this ran against our compiler nightly builds last night and passed.

**Pull request merged**